classes/Settings.js

```diff
@@ -1,13 +1,61 @@
 const { Base64 } = require('js-base64')
+const _ = require('lodash')
 const yaml = require('js-yaml')
 const Bluebird = require('bluebird')
 
 // import classes
 const { Config } = require('../classes/Config.js')
-const { File, DataType } = require('../classes/File.js')
+const { File, DataType, HomepageType } = require('../classes/File.js')
 
 // Constants
 const FOOTER_PATH = 'footer.yml'
+const NAVIGATION_PATH = 'navigation.yml'
+const HOMEPAGE_INDEX_PATH = 'index.md' // Empty string
+
+const retrieveSettingsFiles = async (accessToken, siteName, shouldRetrieveHomepage) => {
+  const configResp = new Config(accessToken, siteName)
+
+  const FooterFile = new File(accessToken, siteName)
+  const dataType = new DataType()
+  FooterFile.setFileType(dataType)
+
+  const NavigationFile = new File(accessToken, siteName)
+  NavigationFile.setFileType(dataType)
+
+  const HomepageFile = new File(accessToken, siteName)
+  const homepageType = new HomepageType()
+  HomepageFile.setFileType(homepageType)
+
+  const fileRetrievalArr = [configResp.read(), FooterFile.read(FOOTER_PATH), NavigationFile.read(NAVIGATION_PATH)]
+
+  // Retrieve homepage only if flag is set to true
+  if (shouldRetrieveHomepage) {
+    fileRetrievalArr.push(HomepageFile.read(HOMEPAGE_INDEX_PATH))
+  }
+
+  const fileContentsArr = await Bluebird.map(fileRetrievalArr, async (fileOp, idx) => {
+    const { content, sha } = await fileOp
+
+    if (idx < 3) {
+      return { content: yaml.safeLoad(Base64.decode(content)), sha}
+    }
+
+    // homepage requires special extraction as the content is wrapped in front matter
+    if (idx === 3) {
+      const homepageContent = Base64.decode(content)
+      const homepageFrontMatterObj = yaml.safeLoad(homepageContent.split('---')[1])
+      return { content: homepageFrontMatterObj, sha }
+    }
+  })
```

Comment:
Nit: Could we consider having fileContentsArr as an object instead, with the appropriate key-value pair e.g. config: { content : ... }

Currently this part of the code is a little harder to understand/care needs to be taken when extending because of the magic number, although I think it's fine to leave it as is if it's difficult to change, since it's not that difficult to understand.

Reply:
good point, resolved with [
d792730](https://github.com/isomerpages/isomercms-backend/pull/85/commits/d792730bc81048e64e6ac12423b2f597309abc3c)